File: app/src/main/java/info/nightscout/androidaps/plugins/pump/danaR/BluetoothDevicePreference.java
package info.nightscout.androidaps.plugins.pump.danaR;

import android.bluetooth.*;
import android.content.Context;
import android.preference.ListPreference;
import android.util.AttributeSet;

import java.util.Set;
import java.util.Vector;

public class BluetoothDevicePreference extends ListPreference {

    public BluetoothDevicePreference(Context context, AttributeSet attrs) {
        super(context, attrs);

        BluetoothAdapter bta = BluetoothAdapter.getDefaultAdapter();
        Vector<CharSequence> entries = new Vector<CharSequence>();
        if (bta != null) {
            Set<BluetoothDevice> pairedDevices = bta.getBondedDevices();
            for (BluetoothDevice dev : pairedDevices) {
                String name = dev.getName();
                if(name != null) {
                    entries.add(name);
                }
            }
        }
        setEntries(entries.toArray(new CharSequence[0]));
        setEntryValues(entries.toArray(new CharSequence[0]));
    }

    public BluetoothDevicePreference(Context context) {
        this(context, null);
    }

}

File: danar/src/main/java/info/nightscout/androidaps/danar/BluetoothDevicePreference.kt
package info.nightscout.androidaps.danar

import kotlin.jvm.JvmOverloads
import android.bluetooth.BluetoothAdapter
import android.bluetooth.BluetoothDevice
import android.bluetooth.BluetoothManager
import android.content.Context
import android.util.AttributeSet
import androidx.preference.ListPreference
import java.util.*

class BluetoothDevicePreference @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) : ListPreference(context, attrs) {
    init {
        val entries = Vector<CharSequence>()
        (context.getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager?)?.adapter?.let { bta ->
            for (dev in bta.bondedDevices)
                dev.name?.let { name -> entries.add(name) }
        }
        setEntries(entries.toTypedArray())
        entryValues = entries.toTypedArray()
    }
}